src/main/java/com/avispl/symphony/dal/infrastructure/management/qsc/qsyscore/device/QSYSPeripheralDevice.java

```diff
@@ -148,4 +149,24 @@ public void updateStatusMessage() {
 		}
 
 	}
+	/**
+	 * Create switch is control property for metric
+	 *
+	 * @param name the name of property
+	 * @param status initial status (0|1)
+	 * @return AdvancedControllableProperty switch instance
+	 */
+	public AdvancedControllableProperty createSwitch(String name, int status, String labelOff, String labelOn) {
```

Comment:
createSwitch method is also present in com.avispl.symphony.dal.util.ControllablePropertyFactory
The only thing is that labels are only On/Off there atm. Please check if it fits your needs :)